pkg/webui/containers/side-bar/index.js

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+import classnames from 'classnames'
+
+import LAYOUT from '@ttn-lw/constants/layout'
+
+import SearchButton from '@ttn-lw/components/search-button'
+
+import SidebarNavigation from './navigation'
+import SidebarContext from './context'
+import SideHeader from './header'
+import SideFooter from './footer'
+import getCookie from './utils'
+import SwitcherContainer from './switcher'
+
+import style from './side-bar.styl'
+
+const getViewportWidth = () =>
+  Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0)
+
+const Sidebar = () => {
+  const viewportWidth = getViewportWidth()
```

Comment:
You are only getting this initially but you should also keep the value updated if the viewport changes.

If you use this more often, you might even consider building a hook for this `const {width } = useViewportDimensions()` that keeps the value updated by hooking into the resize event.

Reply:
If I resize the window now, the sidebar disappears. So that the user can click on the header burger icon to open the drawer. Is this not the intended behavior?

Reply:
Yeah, actually the fact that it measures on every render should be enough in this scenario.